language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -79,22 +79,22 @@
     private int shutdown = 1;
 
     public BallerinaLanguageServer() {
-        this(new BallerinaWorkspaceManager());
+        this(new LanguageServerContextImpl());
     }
 
-    public BallerinaLanguageServer(WorkspaceManager workspaceManager) {
-        super(workspaceManager);
-        this.textService = new BallerinaTextDocumentService(this, workspaceManager);
-        this.workspaceService = new BallerinaWorkspaceService(this, workspaceManager);
-        this.ballerinaDocumentService = new BallerinaDocumentServiceImpl(workspaceManager);
-        this.ballerinaConnectorService = new BallerinaConnectorServiceImpl();
+    private BallerinaLanguageServer(LanguageServerContext serverContext) {
+        super(serverContext);
+        this.textService = new BallerinaTextDocumentService(this, workspaceManager, lsContext);
```

Comment:
Shall we rename lsContext as serverContext? Since we are referring the same here.

Reply:
Done